### ISSUE TITLE
Models that have non-tensor elements in state_dict, are not ONNX-exportable (and not JIT-traceable)

### ISSUE THREAD

**Issue by borisfom** [pytorch/pytorch#135398]
### 🐛 Describe the bug

I had problems exporting a Transformer-based model to ONNX that had some "_extra_state" keys  with None values in the state_dict. Export would break on detach() call below. This simple fix remedied the situation.   

```
--- /usr/local/lib/python3.10/dist-packages/torch/jit/_trace.py 2024-09-05 23:35:21.651452023 +0000
+++ /usr/local/lib/python3.10/dist-packages/torch/jit/_trace.bak        2024-07-09 00:37:50.000000000 +0000
@@ -74,7 +74,7 @@
     filtered_dict = type(state_dict)()
     seen_ids: Set[int] = set()
     for k, v in state_dict.items():
-        if id(v) in seen_ids or v is None:
+        if id(v) in seen_ids:
             continue
         seen_ids.add(id(v))
         if keep_vars:


### Versions

Pytorch nightly

**Comment by justinchuby** [comment]
Please feel free to create a pull request
